text/0091-ci-upload-tokens.md

```diff
@@ -0,0 +1,247 @@
+  automatic fallback.  If a site is not provided, one from the client config
+  should be picked up (typically `sentry.io`).
+* `org`: a token is uniquely bound to an org, so the slug of that org is also always
+  contained.  Note that the slug is used rather than an org ID as the clients typically
+  need these slugs to create API requests.
+* `projects`: a token can be valid for more than one project.  For operations such as
+  source map uploads it's benefitial to issue tokens bound to a single project in which
+  case the upload experience does not require providing the project slugs.
+
+An example of this with a JWT token:
+
+```python
+>>> import jwt
+>>> tok = "sntrys_eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzZW50cnkuaW8iLCJpYXQiOjE2ODQxNTQ2MjYsInNlbnRyeV9zaXRlIjoiaHR0cHM6Ly9teW9yZy5zZW50cnkuaW8vIiwic2VudHJ5X29yZyI6Im15b3JnIiwic2VudHJ5X3Byb2plY3RzIjpbIm15cHJvamVjdCJdfQ.ROnK3f72jGbH2CLkmswMIxXP1qZHDish9lN6kfCR0DU"
+>>> jwt.decode(tok[7:], options={"verify_signature": False})
+{
+  'iss': 'sentry.io',
+  'iat': 1684154626,
+  'sentry_site': 'https://myorg.sentry.io/',
+  'sentry_org': 'myorg',
+  'sentry_projects': ['myproject']
+}
+```
+
+## Token Issuance
+
+The purpose of this change is to allow any organization member to issue tokens with little
+overhead. As users can already issue tokens with shocking levels of access to any of the
+orgs they are a member of there is a lot of room for improvement.
+
+The proposed initial step is to only permit token issuance to support uploads and to permit
+all users in the org to issue such tokens.  The tokens can be shown in the org's
+"Developer Settings" page under a new tab called "Tokens".
+
+Such simple token issuance can then also take place in wizards and documentation pages.
+
+# How To Teach
+
+Structural tokens change what needs to be communicated to users quite a bit.  In particular
+less information is necessary for tools that are compatible with structural tokens.
+This for instance would change this complex webpack config from the docs which requires
+matching `org`, `project` and manually creating a sentry token today:
+
+```javascript
+const SentryWebpackPlugin = require("@sentry/webpack-plugin");
+
+module.exports = {
+  devtool: "source-map",
+  plugins: [
+    new SentryWebpackPlugin({
+      org: "demo-org",
+      project: "demo-project",
+      include: "./dist",
+      // Auth tokens can be obtained from https://sentry.io/settings/account/api/auth-tokens/
+      // and needs the `project:releases` and `org:read` scopes
+      authToken: process.env.SENTRY_AUTH_TOKEN,
+    }),
+  ],
+};
+```
+
+With structural tokens this can be changed to a much more simplified version which also
+correctly handles Single Tenant:
+
+```javascript
+const SentryWebpackPlugin = require("@sentry/webpack-plugin");
+
+module.exports = {
+  devtool: "source-map",
+  plugins: [
+    new SentryWebpackPlugin({
+      authToken: "AUTO GENERATED TOKEN HERE",
```

Comment:
Should this still be something like `process.env.SNTRY_STRUCTURED_TOKEN` for this example? This seems to imply that it is not a secret

Reply:
@evanpurkhiser the idea is that the docs directly inject the token same as they do for the slugs.